src/main/java/dev/carrico/controllers/TagLinkController.java

```diff
@@ -0,0 +1,66 @@
+package dev.carrico.controllers;
+
+import dev.carrico.entities.TagLink;
+import dev.carrico.services.TagLinkService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Set;
+
+@Component
+@RestController
+public class TagLinkController {
+    @Autowired
+    TagLinkService tagLinkService;
+
+    @PostMapping("/tagLinks")
+    @ResponseBody
+    public TagLink createTagLink(@RequestBody TagLink tagLink){
+        this.tagLinkService.createTagLink(tagLink);
+        return tagLink;
+    }
+
+    @GetMapping("/tagLinks/{tagLinkId}")
+    @ResponseBody
+    public TagLink getTagLinkById(@PathVariable int tagLinkId){
+        TagLink tagLink = this.tagLinkService.getTagLinkById(tagLinkId);
+        return tagLink;
+    }
+
+    @GetMapping("/tagLinks")
+    @ResponseBody
+    public Set<TagLink> getAllTagLinks(@RequestParam(name = "tagId", defaultValue = "") String tagId,
+                                       @RequestParam(name = "cardId", defaultValue = "") String cardId){
+        Set<TagLink> tagLinks;
+        if (tagId.isEmpty() && cardId.isEmpty()){
+            tagLinks = this.tagLinkService.getAllTagLinks();
+        }
+        else if (!tagId.isEmpty()){
+            tagLinks = this.tagLinkService.getTagLinksByTagId(Integer.parseInt(tagId));
+        }
+        else if (!cardId.isEmpty()){
+            tagLinks = this.tagLinkService.getTagLinksByCardId(Integer.parseInt(cardId));
+        }
+        else {
+            tagLinks = null;
+        }
+        return tagLinks;
+    }
```

Comment:
There's probs a better way to do this... lets keep this on our TODO to clean up a bit